### PR TITLE
Initial Support for Fujitsu RX300 with iRMC4

### PR DESCRIPTION
**Pull request by Fehler12**
Initial Support for Fujitsu RX300 with iRMC4 (might work for other servers, too)

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [no web gui changes] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: mibs/fujitsu/SC2.mib [hunk 1]
```diff
@@ -0,0 +1,7322 @@
+FSC-SERVERCONTROL2-MIB DEFINITIONS ::= BEGIN
```
Comment (murrant) [inline mibs/fujitsu/SC2.mib]: This mib file should be named `FSC-SERVERCONTROL2-MIB`

**Comment by murrant** [comment]
Looks like your detection is overlapping with another OS `fujitsupyos`  Perhaps this is a duplicate of that OS?

File: mibs/fujitsu/SC2.mib (hunk 1 above)
Comment (Fehler12) [reply mibs/fujitsu/SC2.mib]: I used the same file names as fujitsu. Shall I rename it anyway?

**Comment by Fehler12** [comment]
> Looks like your detection is overlapping with another OS `fujitsupyos` Perhaps this is a duplicate of that OS?

As pyos is a switch and iRMC is an onboard server management module I'd say defenitly no! Why are they overlapping?

File: mibs/fujitsu/SC2.mib (hunk 1 above)
Comment (laf) [reply mibs/fujitsu/SC2.mib]: Yes pls

**Comment by laf** [comment]
> > Looks like your detection is overlapping with another OS `fujitsupyos` Perhaps this is a duplicate of that OS?
> 
> As pyos is a switch and iRMC is an onboard server management module I'd say defenitly no! Why are they overlapping?

Because you are looking for `.1.3.6.1.4.1.231` which is also returned in our test data for `fujitsupyos` so you'll need to update your detection to maybe use sysDescr.

**Comment by Fehler12** [comment]
Changed MIB-Name and added a sysDescr parameter for better detection and to avoid detection collision with pyos. Thank you @laf and @murrant for explaining!

**Comment by Fehler12** [comment]
I think I have found out how to generate valid test data :) and how num_oid works. Is this pull now fine for you?

File: mibs/fujitsu/SC2.mib (hunk 1 above)
Comment (Fehler12) [reply mibs/fujitsu/SC2.mib]: done!

File: includes/definitions/discovery/fujitsuiRMC.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        serial: FSC-SERVERCONTROL2-MIB::sc2UnitSerialNumber.1 # if MIB is not named here, value is not displayed. 
+    sensors:
+        state:
+            data:
+                -
+                    oid: sc2AgentStatus
+                    value: sc2AgentStatus
+                    num_oid: '.1.3.6.1.4.1.231.2.10.2.2.10.8.1.{{ $index }}'
+                    descr: 'Global System Status'
+                    #index: 'statusSummary.{{ $index }}'
```
Comment (laf) [inline includes/definitions/discovery/fujitsuiRMC.yaml:15]: How come you've commented the index key value pair out?

Comment (Fehler12) [reply includes/definitions/discovery/fujitsuiRMC.yaml]: This was a leftover from my tests for a correct yaml and correct test data. Removed this.

**Review by murrant — APPROVED** [review]
Nice work!

**Pull request merged**